corehq/apps/app_manager/commcare_settings.py

```diff
@@ -38,16 +38,21 @@ def _load_custom_commcare_settings():
             if not setting.get('type'):
                 setting['type'] = 'properties'
             settings.append(setting)
-
     with open(os.path.join(path, 'commcare-app-settings.yml'), encoding='utf-8') as f:
         for setting in yaml.safe_load(f):
             if not setting.get('type'):
                 setting['type'] = 'hq'
             settings.append(setting)
+
     for setting in settings:
         if not setting.get('widget'):
             setting['widget'] = 'select'
-
+        if 'values' in setting and 'value_names' not in setting:
+            setting['value_names'] = [titleslug(v) for v in setting['values']]
```

Comment:
My two cents is that this kind of thing (extrapolating the user-facing text from the slug) optimizes for the original writing of the code, rather than the poor person who's trying to find their place in the code later. It's not a big deal either way, but it makes there be more than one way to do things which is now more than one way the person who reads it has to understand when getting into it. I'd suggest just spelling it out every time in the definition.

Reply:
That is a great point. There should be only one way to do things. But I am also compelled by my laziness to offer a lazy variation. Do you think the [`value_pairs` -> `values`](https://github.com/dimagi/commcare-hq/pull/32032/commits/8e017a8ea62af4e8d3dc2c1141d1b2b7b9098e55) commit finds a balance between having only one (hopefully simple) way, and the option of leaving out unnecessary stuff?

My feelings will not be hurt by an "uh, no". I'm happy to drop the stuff that creates value names if you think this could be confusing.

Reply:
Yeah I think either way (before after this latest change) are kind of the same to me. I was thinking along the lines that it's a violation of the "grep rule"—the idea that if you see a string (say, in the UI), you should be able to do an exact string search to find where it is in the codebase. For example, we use that idea when naming our metrics in code—even though all metrics from commcare start with `commcare.`, we still make the caller pass in the full metrics string including the `commcare.`, so that anytime you see a string in datadog, you can search for it in the codebase and find it right away. There are of course times it's worth violating the rule, but it's one I like to keep in mind.

Anyway, that's my feedback, but I'm honestly not very insistent on it in this case. You're closer to the code so you can decide how practically important that is here.

Reply:
OK, that's fair. I'm convinced.